src/elasticsearch_cleanup/main.py

```diff
@@ -0,0 +1,200 @@
+                bulk_actions.append(
+                    {
+                        "_op_type": "delete",
+                        "_index": index,
+                        "_id": doc["_id"],
+                    }
+                )
+            else:
+                seen_values.add(unique_field_value)
+
+    return bulk_actions
+
+
+def fetch_all_documents(
+    opensearch_client: OpenSearch,
+    index_name: str,
+    query: dict[str, Any],
+) -> Generator:
+    """Fetches all documents from an OpenSearch index based on a query.
+
+    Args:
+        opensearch_client: The client to communicate with OpenSearch.
+        index_name: The name of the index.
+        query: The query to filter documents.
+
+    Returns:
+        A generator that yields fetched documents.
+    """
+    return helpers.scan(
+        opensearch_client,
+        index=index_name,
+        query=query,
+        ignore_unavailable=True,
+    )
+
+
+def get_value_from_dict(dot_notation_string: str, data: dict[str, Any]) -> Any:
+    """Retrieves a value from a dictionary based on dot notation string.
+
+    Args:
+        dot_notation_string: The string in dot notation representing keys in the dictionary.
+        data: The dictionary to extract data from.
+
+    Returns:
+        The value extracted from the dictionary based on the dot notation string.
+
+    Raises:
+        ValueError: If any key in the dot notation string is not found in the dictionary.
+    """
+    keys = dot_notation_string.split(".")
+
+    for key in keys:
+        if key in data.keys():
+            data = data[key]
+        else:
+            raise ValueError(f"Key {key} not found in the dictionary")
+
+    return data
+
+
+def main() -> None:
+    if (
+        config.ES_INDEX is None
+        or config.ES_URL is None
+        or config.ES_USER is None
+        or config.ES_PASSWORD is None
+        or config.DOCUMENT_COMPARISON_FIELD is None
+    ):
+        raise ValueError("faled to get all reuqired environment variables")
+
+    opensearch_client = OpenSearch(
+        config.ES_URL,
+        http_auth=(config.ES_USER, config.ES_PASSWORD),
+        verify_certs=False,
+        ssl_show_warn=False,
+    )
+
+    documents = fetch_all_documents(
+        opensearch_client=opensearch_client,
+        index_name=config.ES_INDEX,
+        query=consts.OPENSEARCH_QUERY_ALL_INDEX_DOCUMENTS,
+    )
+
+    log_documents = tee(documents, 1)
+    logger.info(
+        f"Found '{len(list(log_documents))}' documents inside '{config.ES_INDEX}' index"
+    )
+
+    bulk_actions = get_bulk_actions(
+        documents=documents,
+        comparison_field=config.DOCUMENT_COMPARISON_FIELD,
+        index=config.ES_INDEX,
+        logger=logger,
+    )
+    logger.info(f"Found '{len(bulk_actions)}' duplicated documents")
+
+    if len(bulk_actions) == 0:
+        logger.info(f"No duplicates found in '{config.ES_INDEX}'.")
+        return
+
+    dry_run_mode: bool = str_to_bool(config.DRY_RUN)
```

Comment:
```suggestion
    dry_run_mode = str_to_bool(config.DRY_RUN)
```

you can instead annotate the function itself